deepchem/models/dft/dftxc.py

```diff
@@ -0,0 +1,192 @@
+
+
+class XCModel(TorchModel):
+    """
+    This class is used to initialize and run Differentiable Quantum Chemistry (i.e,
+    DFT) calculations, using an exchange correlation functional that has been replaced
+    by a neural network. This model is based on the paper "Learning the exchange-correlation
+    functional from nature with fully differentiable density functional
+    theory." and is listed below for reference.
+
+    To read more about Density Functional Theory and the exchange
+    correlation functional please check the references below.
+
+    Examples
+    --------
+    >>> from deepchem.models.dft.dftxc import XCModel
+    >>> from deepchem.data.data_loader import DFTYamlLoader
+    >>> inputs = 'deepchem/models/tests/assets/test_dftxcdata.yaml'
+    >>> data = DFTYamlLoader()
+    >>> dataset = (data.create_dataset(inputs))
+    >>> model = XCModel("lda_x", batch_size=1)
+    >>> loss = model.fit(dataset, nb_epoch=1, checkpoint_interval=1)
+
+    Notes
+    -----
+    The entry type "Density Matrix" cannot be used on model.evaluate as of now.
+    To run predictions on this data type, a dataset containing only "dm" entries must
+    be used.
+
+    References
+    ----------
+    deepchem.models.dft.nnxc
+    Encyclopedia of Condensed Matter Physics, 2005.
+    Kasim, Muhammad F., and Sam M. Vinko. "Learning the exchange-correlation
+    functional from nature with fully differentiable density functional
+    theory." Physical Review Letters 127.12 (2021): 126403.
+    """
+
+    def __init__(self,
+                 xcstr: str,
+                 ninp: int = 2,
+                 nhid: int = 10,
+                 ndepths: int = 1,
+                 modeltype: int = 1,
+                 n_tasks: int = 0,
+                 log_frequency: int = 0,
+                 mode: str = 'classification',
+                 device: Optional[torch.device] = None,
+                 **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        xcstr: str
+            The choice of xc to use.
+        ninp: int
+            size of neural input
+        nhid: int
+            hidden layer size
+        ndepths: int
+            depth of neural network
+        modeltype: int
+            model type 2 includes an activation layer whereas type 1 does not.
+        """
+
+        model = DFTXC(xcstr, ninp, nhid, ndepths, modeltype)
+        self.xc = xcstr
+        self.model = model
+        loss: Loss = L2Loss()
+        output_types = ['loss', 'predict']
+        self.mode = mode
+        super(XCModel, self).__init__(model,
+                                      loss=loss,
+                                      output_types=output_types,
+                                      **kwargs)
+
+    def _prepare_batch(self, batch):
+
+        inputs, labels, weights = batch
+        labels = [torch.from_numpy(inputs[0][0].get_true_val())]
+        labels[0].requires_grad_()
+        w = np.array([1.0])
```

Comment:
Let's add a unit test for the case when `batch_size` does not equal 1. I think it may ignore the other elements in the batch in that case